### ISSUE TITLE
Wrong player preserve spectator state after disconnect

### ISSUE THREAD

**Issue by M0REKZ** [ddnet-insta/ddnet-insta#192]
Try the next:
* Join server
* Go spectators
* Disconnect
* Rejoin
* You are still spectator

Sometimes it's kinda annoying and confuses other players

**Comment by ChillerDragon** [comment]
I can not reproduce.

**Comment by M0REKZ** [comment]
Sorry i found this bug on my fork so maybe its just fork bug (even if i didnt touch spectator code), however now i cant reproduce it even on my fork xD which makes it more weird so... i guess ill just close this?

**Comment by ChillerDragon** [comment]
Yeah lets close it. We can reopen it if it happens again. But I have never seen it so far.

**Comment by M0REKZ** [comment]
Found a consistent way to reproduce the bug:

* Open 2 DDNet clients
* Join first client
* Join second
* Join first client dummy
* Join second client dummy
* Make FIRST client dummy spectator
* Change map
* Join SECOND client dummy
* Join First client dummy
* Wrong client is spectator

**Comment by ChillerDragon** [comment]
I did not check the code yet or try to reproduce but I would assume this is a ddnet bug. There is persisted player data that is kept across map reloads and it holds the is spectator state.

**Comment by ChillerDragon** [comment]
Thanks for the detailed reproduce steps. Works nice!

I could reproduce the bug in ddnet so it is out of scope for ddnet-insta. Fixing this in ddnet-insta only causes too much code diff that makes merging harder. This is a bug that should be fixed in ddnet. If you need this fixed please open an issue in the ddnet repo.

**Comment by ChillerDragon** [comment]
Now that we have https://github.com/ddnet-insta/ddnet-insta/commit/fe75a7f065c829990660b2defe7e0f66d679d204 this can be revisited. This should be solvable without introducing more diff to ddnet now.

**Comment by ChillerDragon** [comment]
Looks like this already got fixed upstream https://github.com/ddnet/ddnet/issues/9666
